packages/component-library/src/utils/stores.ts

```diff
@@ -0,0 +1,73 @@
+import { ref } from "vue";
+import { defineStore } from "pinia";
+import type { TokenHandler } from "./auth/tokenhandler.types";
+import { oauthFactory } from "./auth/oauth";
+import { credentialsAuthFactory } from "./auth/credentials";
+
+export function authStoreFactory(
+  tokenHandler: TokenHandler<{ access: string; refresh: string }>,
+  baseUrl?: string
+) {
+  const _oauth = oauthFactory(tokenHandler, baseUrl);
+  const _credentials = credentialsAuthFactory(tokenHandler, baseUrl);
+
+  return defineStore("alerceAuth", () => {
+    const loggedUser = ref({
+      isLogged: false,
+      user: "",
+    });
+    // add alerts when logout failed :D
+
+    async function oauthLoginPopup(callbackUrl: string) {
+      const url = await _oauth.getUrl(callbackUrl);
+      return window.open(url);
+    }
+
+    function oauthLogin(code: string, state: string) {
+      return _oauth.signIn(code, state);
+    }
+
+    async function credentialsLogin(username: string, password: string) {
+      try {
+        await _credentials.signIn(username, password);
+        loggedUser.value = {
+          isLogged: true,
+          user: username,
+        };
+      } catch (e) {
+        console.error("Invalid username or password");
```

Comment:
Shouldn't we just throw errors at this point so the application can handle them ?

Reply:
true